internal/services/recoveryservices/site_recovery_replicated_vm_resource.go

```diff
@@ -1059,6 +1127,84 @@ func waitForReplicationToBeHealthyRefreshFunc(d *pluginsdk.ResourceData, meta in
 	}
 }
 
+func expandSiteRecoveryReplicatedVMIPConfig(nicInput map[string]interface{}) []replicationprotecteditems.IPConfigInputDetails {
+	output := []replicationprotecteditems.IPConfigInputDetails{}
+	ipConfigs := nicInput["ip_configuration"].([]interface{})
+	if len(ipConfigs) > 0 {
+		for _, ipConfig := range ipConfigs {
+			ipConfig := ipConfig.(map[string]interface{})
+			var recoveryLoadBalancerBackendPoolIds *[]string
+			if ids, ok := nicInput["recovery_load_balancer_backend_address_pool_ids"].(*schema.Set); ok && ids.Len() > 0 {
+				recoveryLoadBalancerBackendPoolIds = utils.ExpandStringSlice(ids.List())
+			}
+			output = append(output, replicationprotecteditems.IPConfigInputDetails{
+				IPConfigName:                    pointer.To(ipConfig["name"].(string)),
+				RecoverySubnetName:              pointer.To(ipConfig["target_subnet_name"].(string)),
+				RecoveryStaticIPAddress:         pointer.To(ipConfig["target_static_ip"].(string)),
+				RecoveryLBBackendAddressPoolIds: recoveryLoadBalancerBackendPoolIds,
+				RecoveryPublicIPAddressId:       pointer.To(ipConfig["recovery_public_ip_address_id"].(string)),
+				TfoStaticIPAddress:              pointer.To(ipConfig["failover_test_static_ip"].(string)),
+				TfoSubnetName:                   pointer.To(ipConfig["failover_test_subnet_name"].(string)),
+				TfoPublicIPAddressId:            pointer.To(ipConfig["failover_test_public_ip_address_id"].(string)),
+				IsPrimary:                       pointer.To(ipConfig["is_primary"].(bool)),
+			})
+		}
+		return output
+	}
+
+	if !features.FivePointOh() {
+		targetStaticIp := nicInput["target_static_ip"].(string)
+		targetSubnetName := nicInput["target_subnet_name"].(string)
+		recoveryPublicIPAddressID := nicInput["recovery_public_ip_address_id"].(string)
+		testStaticIp := nicInput["failover_test_static_ip"].(string)
+		testSubNetName := nicInput["failover_test_subnet_name"].(string)
+		testPublicIpAddressID := nicInput["failover_test_public_ip_address_id"].(string)
+
+		var recoveryLoadBalancerBackendPoolIds *[]string
+		if ids, ok := nicInput["recovery_load_balancer_backend_address_pool_ids"].(*schema.Set); ok && ids.Len() > 0 {
+			recoveryLoadBalancerBackendPoolIds = utils.ExpandStringSlice(ids.List())
+		}
+
+		if targetStaticIp != "" || targetSubnetName != "" || recoveryPublicIPAddressID != "" || testSubNetName != "" || testStaticIp != "" || testPublicIpAddressID != "" {
+			return append(output, replicationprotecteditems.IPConfigInputDetails{
+				RecoverySubnetName:              &targetSubnetName,
+				RecoveryStaticIPAddress:         &targetStaticIp,
+				RecoveryLBBackendAddressPoolIds: recoveryLoadBalancerBackendPoolIds,
+				RecoveryPublicIPAddressId:       &recoveryPublicIPAddressID,
+				TfoStaticIPAddress:              &testStaticIp,
+				TfoPublicIPAddressId:            &testPublicIpAddressID,
+				TfoSubnetName:                   &testSubNetName,
+				IsPrimary:                       utils.Bool(true),
```

Comment:
`utils.Bool` is deprecated

```suggestion
				IsPrimary:                       pointer.To(true),
```